sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,427 @@
+
+        // Add the "Content-Id" header which allows this request to be mapped to the response.
+        context.getHttpRequest().setHeader(CONTENT_ID, context.getData(BATCH_REQUEST_CONTENT_ID).get().toString());
+
+        return next.process();
+    }
+
+    /*
+     * This performs changing the request URL to the value passed through the pipeline context. This policy is used in
+     * place of constructing a new client for each batch request that is being sent.
+     */
+    private Mono<HttpResponse> setRequestUrl(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // Set the request URL to the correct endpoint.
+        try {
+            UrlBuilder requestUrl = UrlBuilder.parse(context.getHttpRequest().getUrl());
+            requestUrl.setPath(context.getData(BATCH_REQUEST_URL_PATH).get().toString());
+            context.getHttpRequest().setUrl(requestUrl.toURL());
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(Exceptions.propagate(new IllegalStateException(ex)));
+        }
+
+        return next.process();
+    }
+
+    /*
+     * This will "send" the batch operation request when triggered, it simply acts as a way to build and write the
+     * batch operation into the overall request and then returns nothing as the response.
+     */
+    private Mono<HttpResponse> setupBatchOperation(HttpRequest request) {
+        int contentId = Integer.parseInt(request.getHeaders().remove(CONTENT_ID).getValue());
+
+        StringBuilder batchRequestBuilder = new StringBuilder();
+        appendWithNewline(batchRequestBuilder, "--" + batchBoundary);
+        appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TYPE);
+        appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TRANSFER_ENCODING);
+        appendWithNewline(batchRequestBuilder, String.format(BATCH_OPERATION_CONTENT_ID_TEMPLATE, contentId));
+        batchRequestBuilder.append(BlobBatchHelper.HTTP_NEWLINE);
+
+        String method = request.getHttpMethod().toString();
+        String urlPath = request.getUrl().getPath();
+        String urlQuery = request.getUrl().getQuery();
+        if (!ImplUtils.isNullOrEmpty(urlQuery)) {
+            urlPath = urlPath + "?" + urlQuery;
+        }
+        appendWithNewline(batchRequestBuilder, String.format(OPERATION_TEMPLATE, method, urlPath, HTTP_VERSION));
+
+        request.getHeaders().stream()
+            .filter(header -> !X_MS_VERSION.equalsIgnoreCase(header.getName()))
+            .forEach(header -> appendWithNewline(batchRequestBuilder,
+                String.format(HEADER_TEMPLATE, header.getName(), header.getValue())));
+
+        batchRequestBuilder.append(BlobBatchHelper.HTTP_NEWLINE);
+
+        batchRequest.add(ByteBuffer.wrap(batchRequestBuilder.toString().getBytes(StandardCharsets.UTF_8)));
+        batchMapping.get(contentId).setRequest(request);
+
+        return Mono.empty();
```

Comment:
I would replace this with 
```java
return Mono.fromRunnable(() -> {
// stuff
});
```
So that the string concatonation and so forth can happen asynchronously.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756